academia/curriculum/learning_task.py

```diff
@@ -303,11 +303,11 @@ def __run_episode(self, agent: Agent, evaluation_mode: bool = False) -> tuple[fl
 
             if not evaluation_mode:
                 agent.update(state, action, reward, new_state, done)
-                agent.update_exploration()
 
             state = new_state
             episode_reward += reward
             steps_count += 1
+        agent.update_exploration()
```

Comment:
this should be placed inside the same `if` block as before since we don't want to update exploration in evaluation mode